### PR TITLE
Post-zed/2967 "put" doc updates

### PR DESCRIPTION
**Pull request by philrz**
Now that #2967 is merged, I re-read the relevant docs as a user and noticed a couple adjustments that seem appropriate.

### CODE REVIEW

File: docs/language/operators/README.md [hunk 1]
```diff
@@ -685,8 +685,8 @@ time                        uid
 
 |                           |                                                 |
 | ------------------------- | ----------------------------------------------- |
-| **Description**           | Add/update fields based on the results of an expression         |
-| **Syntax**                | `put <field> := <expression> [, <field> := <expression> ...]`   |
+| **Description**           | Add/update fields based on the results of an expression<br><br>The `put` keyword is optional, though its explicit use may be appropriate to disambiguate from where `:=` is used for [naming fields to hold aggregation results](../aggregate-functions/README.md#field-naming). |
```
Comment (philrz) [inline docs/language/operators/README.md:688]: @mattnibs @nwt @mccanne: Please keep me honest here! I _think_ I've listed the only other context in the language where `:=` comes up, but I sometimes have difficulty keeping it all mentally cached.

Comment (mattnibs) [reply docs/language/operators/README.md:688]: No you do not need to use `:=`. For implied puts the semantic compiler is able to figure out if a call is a normal function (i.e. a call in a put) or an aggregation call (i.e. part of summarize proc).

File: docs/language/operators/README.md [hunk 2]
```diff
@@ -685,8 +685,8 @@ time                        uid
...
+| **Syntax**                | `[put] <field> := <expression> [, <field> := <expression> ...]`   |
```
Comment (mattnibs) [inline docs/language/operators/README.md:689]: You can also just have a call here where a field with the name of the call is create: `to_lower(x)`

File: docs/language/operators/README.md [hunk 3]
```diff
@@ -709,6 +709,25 @@ id.orig_h     id.orig_p id.resp_h       id.resp_p orig_bytes resp_bytes total_by
 ...
 ```
 
+#### Example #2:
+
+As noted above, the `put` keyword is optional. Our prior example could be
+abbreviated as shown below and return the same result.
+
+```mdtest-command zed-sample-data/zeek-default
+zq -q -f table 'total_bytes := orig_bytes + resp_bytes | sort -r total_bytes | cut id, orig_bytes, resp_bytes, total_bytes' conn.log.gz
```
Comment (mattnibs) [inline docs/language/operators/README.md:718]: 👏

File: docs/language/operators/README.md (hunk 2 above)
Comment (mattnibs) [reply docs/language/operators/README.md:689]: So a put is a list of either `<field> := <expression>` or `<call_expression>`

**Comment by philrz** [comment]
Thanks for the feedback @mattnibs. All good points. I've pushed a new commit that attempts to incorporate your feedback. Please give it another look and let me know what you think. I confess I sometimes struggle to summarize these slightly-more-sophisticated syntaxes without going into a full narrative trying to explain how the parser works.

**Comment by mattnibs** [comment]
> I confess I sometimes struggle to summarize these slightly-more-sophisticated syntaxes without going into a full narrative trying to explain how the parser works.

Hmm yeah. I have been thinking about this the last couple weeks while I've been doing more work on the grammar. We've got a lot shorthand tricks to allow you to type less in certain situations, but part of me wonders if this makes the language more difficult to grasp. Does the language have too many secret handshakes? Would be it better to design for slightly more typing but more consistency (e.g. making assignments require a left hand `x:=`)?

**Review by mattnibs — APPROVED** [review]

**Pull request merged**